### PR TITLE
We need volunteers for test compatible with the app trucker

### PR DESCRIPTION
**Pull request by darknight2590**

### CODE REVIEW

**Comment by darknight2590** [comment]
add this to player.lua
```
	PlayerData.metadata["jobtier"] = PlayerData.metadata["jobtier"] ~= nil and PlayerData.metadata["jobtier"] or {
		["trucker"] = 0,
	}
```
add this under PlayerData.metadata["phonedata]
```
	PlayerData.metadata["payslips"] = PlayerData.metadata["payslips"] ~= nil and PlayerData.metadata["payslips"] or {
		["trucker"] = 0
	}
```

add this in config.lua in qbcore
```

QBConfig.Tiers = {}
QBConfig.Tiers[1] = 100
QBConfig.Tiers[2] = 200
QBConfig.Tiers[3] = 300
QBConfig.Tiers[4] = 400
QBConfig.Tiers[5] = 500
```

File: client/main.lua [hunk 1]
```diff
@@ -193,17 +195,23 @@ Citizen.CreateThread(function()
...
+												TriggerServerEvent("qb-trucker:server:deliveryDone", CurrentLocation.dropcount)
                                                 QBCore.Functions.Notify("You Have Delivered All Products, To The Next Point")
                                                 local chance = math.random(1,100)
                                                 if chance < 26 then
                                                     TriggerServerEvent('qb-trucker:server:nano')
                                                 end
                                                 if CurrentBlip ~= nil then
                                                     RemoveBlip(CurrentBlip)
-						    ClearAllBlipRoutes()
+													ClearAllBlipRoutes()
```
Comment (FreakOutModz) [inline client/main.lua:214]: Why did you add so many spaces ? Please make the code look clean

File: client/main.lua [hunk 2]
```diff
@@ -234,28 +242,29 @@ Citizen.CreateThread(function()
...
+    TriggerServerEvent("qb-trucker:server:workingLocation", location)
+    if (location == false) then
+        CurrentLocation = nil;
+        return 
     end
+    CurrentLocation = {}
+    CurrentLocation.id = location
+    CurrentLocation.dropcount = math.random(1, 3)
+    CurrentLocation.store = Config.Locations["stores"][location].name
+    CurrentLocation.x = Config.Locations["stores"][location].coords.x
```
Comment (FreakOutModz) [inline client/main.lua:254]: You can do:
`CurrentLocation.xyz = Config.Locations["stores"][location].coords.xyz`

File: client/main.lua [hunk 3]
```diff
@@ -380,3 +394,26 @@ function DrawText3D(x, y, z, text)
...
 end
+
+function GetJobInfo()
+    local returnn = {
+        ['WorkingLocations'] = WorkingLocations,
+        ['JobsDone'] = JobsDone,
+        ['Locations'] = Config.Locations["stores"],
+        ['LocationsDone'] = LocationsDone,  
+    }
+    return returnn
```
Comment (FreakOutModz) [inline client/main.lua:405]: returnn ? What... please name this correctly. That doesn't mean anything.

File: client/main.lua [hunk 4]
```diff
@@ -193,17 +195,23 @@ Citizen.CreateThread(function()
...
-                                            if currentCount == CurrentLocation.dropcount then
+                                            if currentCount >= CurrentLocation.dropcount then
+												local locationId = CurrentLocation.id;
                                                 table.insert(LocationsDone, CurrentLocation.id)
+                                                Citizen.SetTimeout(10 * 1000 * 60, function()
+                                                    table.remove(LocationsDone, locationId)
+                                                    TriggerEvent("qb-trucker:client:locationsDoneUpdate")
+                                                end)												
                                                 TriggerServerEvent("qb-shops:server:RestockShopItems", CurrentLocation.store)
+												TriggerServerEvent("qb-trucker:server:deliveryDone", CurrentLocation.dropcount)
```
Comment (FreakOutModz) [inline client/main.lua:206]: So many spaces here too, plese make a clean code.

File: client/main.lua [hunk 5]
```diff
@@ -193,17 +195,23 @@ Citizen.CreateThread(function()
                                             ClearPedTasks(PlayerPedId())
                                             hasBox = false
                                             currentCount = currentCount + 1
-                                            if currentCount == CurrentLocation.dropcount then
+                                            if currentCount >= CurrentLocation.dropcount then
+												local locationId = CurrentLocation.id;
```
Comment (FreakOutModz) [inline client/main.lua:199]: Same for spaces here too, please correct all the code.

**Review by FreakOutModz — CHANGES_REQUESTED** [review]
All variables should also start with a lower case and each additional word in upper case for a cleaner code.

**Pull request closed without merging**